### PR TITLE
RavenDB-10299: /proc/meminfo + /proc/self/status EPs + commitCharge c…

### PR DESCRIPTION
**Pull request by aviviadi**
…alc for lowmem

### CODE REVIEW

File: src/Raven.Server/Documents/Handlers/Debugging/MemoryStatsHandler.cs [hunk 1]
```diff
@@ -136,14 +139,14 @@ private static DynamicJsonValue MemoryStatsInternal()
...
                     {
-                        var details = new DynamicJsonValue();
-                        
                         var dir = new DynamicJsonValue
                         {
-                            [nameof(MemoryInfoMappingItem.Directory)] = sizes.Key.Substring(prefixLength),
-                            [nameof(MemoryInfoMappingItem.TotalDirectorySize)] = sizes.Value,
-                            [nameof(MemoryInfoMappingItem.HumaneTotalDirectorySize)] = Size.Humane(sizes.Value),
-                            [nameof(MemoryInfoMappingItem.Details)] = details
+                            ["Directory"] = sizes.Key.Substring(prefixLength),
```
Comment (arekpalinski) [inline src/Raven.Server/Documents/Handlers/Debugging/MemoryStatsHandler.cs]: Why did you remove all `nameof()` usages? 
Why did you remove `MemoryInfo*` classes? 
Build does not compile: 
`00:06:07.719 Program.cs(351,56): error CS0426: The type name 'MemoryInfo' does not exist in the type 'MemoryStatsHandler' [c:\Jenkins-lite\workspace\PR_Compile_v4.0\s\tools\TypingsGenerator\TypingsGenerator.csproj]
`

File: src/Raven.Server/Documents/Handlers/Debugging/MemoryStatsHandler.cs [hunk 2]
```diff
@@ -302,58 +308,132 @@ public static int LongestCommonPrefixLength(List<string> strings)
...
+
+        [RavenAction("/admin/debug/proc/meminfo", "GET", AuthorizationStatus.DatabaseAdmin, IsDebugInformationEndpoint = true)]
+        public Task ProcMemInfo()
         {
-            public long FileSize { get; set; }
-            public string HumaneFileSize { get; set; }
-            public long TotalMapped { get; set; }
-            public string HumaneTotalMapped { get; set; }
-            public MemoryInfoMappingDetails[] Mappings { get; set; }
+            var txt = File.ReadAllLines("/proc/meminfo");
```
Comment (arekpalinski) [inline src/Raven.Server/Documents/Handlers/Debugging/MemoryStatsHandler.cs]: Shoudn't we check we are running on Linux first? It's debug endpoint but still shound't crash on windows

File: src/Raven.Server/Documents/Handlers/Debugging/MemoryStatsHandler.cs [hunk 3]
```diff
@@ -302,58 +308,132 @@ public static int LongestCommonPrefixLength(List<string> strings)
...
+                    if (parsedLine.Length != 3) // format should be: {filter}: <num> kb
+                        continue; // an error but we ignore and do not put in the results
+
+                    var key = parsedLine[0].Split(":")[0];
+
+                    if (long.TryParse(parsedLine[1], out var result) == false)
+                    {
+                        djv[key] = "Failed To Parse Value : " + parsedLine[1];
+                        continue;
+                    }
```
Comment (arekpalinski) [inline src/Raven.Server/Documents/Handlers/Debugging/MemoryStatsHandler.cs:401]: For me this piece of code is fragile. Maybe let's do it this way but also include the raw line. Also if we get exception here we'll still be able to output raw message.

File: src/Raven.Server/Documents/Handlers/Debugging/MemoryStatsHandler.cs [hunk 4]
```diff
@@ -302,58 +308,132 @@ public static int LongestCommonPrefixLength(List<string> strings)
...
+
+        [RavenAction("/admin/debug/proc/self/status", "GET", AuthorizationStatus.DatabaseAdmin, IsDebugInformationEndpoint = true)]
+        public Task ProcSelfStatus()
         {
-            public long Size { get; set; }
-            public long Count { get; set; }
+            using (var currentProcess = Process.GetCurrentProcess())
+            {
+                var path = $"/proc/{currentProcess.Id}/status";
+                var txt = File.ReadAllLines(path);
```
Comment (arekpalinski) [inline src/Raven.Server/Documents/Handlers/Debugging/MemoryStatsHandler.cs]: Check if such file exists.

File: src/Raven.Server/Documents/Handlers/Debugging/MemoryStatsHandler.cs [hunk 5]
```diff
@@ -302,58 +308,132 @@ public static int LongestCommonPrefixLength(List<string> strings)
...
+                        {
+                            var restOfTheLine = new StringBuilder(parsedLine[1]);
+                            for (int i = 2; i < parsedLine.Length; i++)
+                            {
+                                restOfTheLine.Append(" ");
+                                restOfTheLine.Append(parsedLine[i]);
+                            }
+                            djv[key] = restOfTheLine.ToString();
+                        }
+                    }
```
Comment (arekpalinski) [inline src/Raven.Server/Documents/Handlers/Debugging/MemoryStatsHandler.cs:506]: The same comment as above. Very fragile code. Let's consider to output raw lines as well.

File: src/Sparrow/LowMemory/MemoryInformation.cs [hunk 6]
```diff
@@ -77,73 +78,98 @@ public struct MemoryStatusEx
...
+                        break;
+                    case "mb":
+                        result *= 1024L * 1024;
+                        break;
+                    case "gb":
+                        result *= 1024L * 1024 * 1024;
+                        break;
+                }
+
+                rcValues[cnt++] = result;
```
Comment (arekpalinski) [inline src/Sparrow/LowMemory/MemoryInformation.cs:162]: Isn't the parsing code the same as in the debug endpoint handler? Can we extract it and have it in one place?

File: src/Sparrow/LowMemory/MemoryInformation.cs [hunk 7]
```diff
@@ -77,73 +78,98 @@ public struct MemoryStatusEx
...
-                case "mb":
-                    result *= 1024L * 1024;
-                    break;
-                case "gb":
-                    result *= 1024L * 1024 * 1024;
-                    break;
+                var found = false;
+                foreach (var filter in filters)
+                {
+                    if (parsedLine[0].Contains($"{filter}:") == true)
```
Comment (arekpalinski) [inline src/Sparrow/LowMemory/MemoryInformation.cs:131]: Is it ok to be case sensitive here?

File: src/Sparrow/Platform/Posix/KernelVirtualFileSystemUtils.cs [hunk 8]
```diff
@@ -206,22 +206,30 @@ public static HashSet<string> GetAllDisksFromPartitionsFile()
...
-                    if (line.Contains(filter))
+                    foreach (var filter in filters)
                     {
-                        result = line;
-                        if (++cnt > 1)
-                            return null;
+                        if (line.Contains(filter))
+                        {
+                            if (cnt > filters.Length - 1)
+                                return null; // TODO: this is an error
```
Comment (arekpalinski) [inline src/Sparrow/Platform/Posix/KernelVirtualFileSystemUtils.cs:223]: What are we going to do with this?

File: src/Raven.Server/Documents/Handlers/Debugging/MemoryStatsHandler.cs (hunk 1 above)
Comment (aviviadi) [reply src/Raven.Server/Documents/Handlers/Debugging/MemoryStatsHandler.cs]: I didn't.. merge issue

Comment (ml054) [reply src/Raven.Server/Documents/Handlers/Debugging/MemoryStatsHandler.cs]: Did you have conflict? Please notice I've also changed format of endpoint output, to avoid mixing data with structure.

File: src/Raven.Server/Documents/Handlers/Debugging/MemoryStatsHandler.cs [hunk 9]
```diff
@@ -348,12 +355,168 @@ internal class MemoryInfoMappingFileInfo
...
         }
-        
+
         internal class MemoryInfoMappingDetails
         {
             public long Size { get; set; }
             public long Count { get; set; }
         }
+        
+        [RavenAction("/admin/debug/proc/meminfo", "GET", AuthorizationStatus.DatabaseAdmin, IsDebugInformationEndpoint = true)]
```
Comment (ayende) [inline src/Raven.Server/Documents/Handlers/Debugging/MemoryStatsHandler.cs:365]: Only cluster operator here.

File: src/Raven.Server/Documents/Handlers/Debugging/MemoryStatsHandler.cs [hunk 10]
```diff
@@ -348,12 +355,168 @@ internal class MemoryInfoMappingFileInfo
...
+                        {
+                            ["ERROR"] = "End point doesn't support non linux OS",
+                        });
+                    }
+                }
+                return Task.CompletedTask;
+            }
+            
+            var txt = File.ReadAllLines("/proc/meminfo");
+            using (ServerStore.ContextPool.AllocateOperationContext(out JsonOperationContext context))
```
Comment (ayende) [inline src/Raven.Server/Documents/Handlers/Debugging/MemoryStatsHandler.cs:384]: We just need the raw data here, nothing more.

File: src/Sparrow/LowMemory/MemoryInformation.cs (hunk 7 above)
Comment (ayende) [reply src/Sparrow/LowMemory/MemoryInformation.cs:131]: Yes, we can expect it to be stable. But it is called a lot and we don't want to pay all these costs

**Comment by ayende** [comment]
@maximburyak is going to work on this as separate stages.
